BeeSwift/GoalViewController.swift

```diff
@@ -368,12 +371,14 @@ class GoalViewController: UIViewController,  UIScrollViewDelegate, DatapointTabl
         var components = DateComponents()
         components.day = Int(self.dateStepper.value)
 
-        let newDate = (calendar as NSCalendar?)?.date(byAdding: components, to: Date(), options: [])
+        let now = Date()
```

Comment:
This does surprising things if you have the view open and midnight rolls around. But this PR doesn't change that weird behavior, it was always there.

Reply:
True. The app will work with the data it has from the time it last fetched it. One can use other means to adjust the deadline and the one client might not know about it while it is at the timer screen, add data screen, or edit data screen.

Is the server posting to the app via remote notification that a goal has been updated? Does the app still need to fetch (pull) to find out?